### PR TITLE
feat!: update to Remix 2

### PR DESCRIPTION
**Pull request by ascorbic**
## Description

Updates the templates for Remix 2. This includes:

- converting the site to ESM
- upgrading Remix
- switching the adapters to the new versions
- new server.ts files and config files, including converting to Functions v2
- replace `entry.server.ts` with a re-export from the adapters
- many changes to the init script, including fixing a bug that prevented git from being initted and another that prevented new dependencies from being installed

### CODE REVIEW

File: remix.init/entry.server.tsx
Comment (ascorbic) [inline remix.init/entry.server.tsx:1]: This file is based on the new default from the remix repo

File: remix.init/netlify-edge-toml [hunk 1]
```diff
@@ -2,6 +2,11 @@
 command = "npm run build"
 publish = "public"
 
-[dev]
-command = "npm run dev"
```
Comment (ascorbic) [inline remix.init/netlify-edge-toml:6]: We no longer run a dev command - ntl is spawned by remix

File: remix.init/root.tsx
Comment (ascorbic) [inline remix.init/root.tsx:1]: Updated to match default

File: remix.init/netlify-toml [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
 publish = "public"
 
 [dev]
 command = "npm run dev"
-targetPort = 3000
+
+[[redirects]]
+from = "/*"
+to = "/.netlify/functions/server"
+status = 200
```
Comment (ascorbic) [inline remix.init/netlify-toml:11]: For now we still need this, because ISC can't do shadowing

File: app/entry.client.tsx
Comment (ascorbic) [inline app/entry.client.tsx:1]: This is the new default in Remix

File: app/entry.server.tsx [hunk 3]
```diff
@@ -1,21 +1 @@
...
-  );
-
-  responseHeaders.set("Content-Type", "text/html");
-
-  return new Response("<!DOCTYPE html>" + markup, {
-    headers: responseHeaders,
-    status: responseStatusCode,
-  });
-}
+export { handleRequest as default } from "@netlify/remix-adapter";
```
Comment (ascorbic) [inline app/entry.server.tsx:1]: We now define the handler in the adapter, so this is all that's needed!

File: remix.init/README-edge.md [hunk 4]
```diff
@@ -40,26 +40,26 @@ npm install
 Run
 
 ```sh
-netlify dev
+npm run dev
```
Comment (ascorbic) [inline remix.init/README-edge.md:43]: We use the new v2 dev mode, which allows remiox dev to run a custom server, which in our case is `ntl dev`. For that reason we dont want them running it themselves

File: remix.init/index.js [hunk 5]
```diff
@@ -3,28 +3,23 @@ const fs = require("fs/promises");
 const { join } = require("path");
 const PackageJson = require("@npmcli/package-json");
 const execa = require("execa");
-const { Command } = require('commander');
+const { Command } = require("commander");
 
-const foldersToExclude = [".github", ".git"];
+const foldersToExclude = [".github"];
```
Comment (ascorbic) [inline remix.init/index.js:8]: The `.git` dir is created by Remix, and is this site's initial commit

File: remix.init/index.js [hunk 6]
```diff
@@ -116,6 +125,7 @@ async function main({ rootDirectory }) {
       rootDirectory,
     });
     await updatePackageJsonForFunctions(rootDirectory);
+    await installAdditionalDependencies({ rootDirectory, packageManager });
```
Comment (ascorbic) [inline remix.init/index.js:128]: Both runtimes need an extra install step

File: server.ts [hunk 7]
```diff
@@ -1,7 +1,9 @@
...
 import * as build from "@remix-run/dev/server-build";
+import { createRequestHandler } from "@netlify/remix-adapter";
 
-export const handler = createRequestHandler({
+const handler = createRequestHandler({
   build,
   mode: process.env.NODE_ENV,
 });
+
+export default handler;
```
Comment (ascorbic) [inline server.ts:9]: It doesn't recognise it as a v2 function if we export it directly, so we assign to a var and export that

**Review by orinokai — APPROVED** [review]
LGTM! 🚢

**Pull request merged**